src/Request.ts

```diff
@@ -267,19 +261,60 @@ export default class Request {
 
       // Fields related to routing:
       this.baseUrl = baseURL;
-      this.path = this.url = event.path;
-      this.originalUrl = baseURL + event.path;
+      this._path = this._url = path;
+      // Despite the fact that the Express docs say that the `originalUrl` is `baseUrl +
+      // path`, it's actually always equal to the original URL that initiated the request.
+      // If, for example, a route handler changes the `url` of a request, the `path` is
+      // changed too, *but* `originalUrl` stays the same. This would not be the case if
+      // `originalUrl = `baseUrl + path`.
```

Comment:
Does this mean that our documentation on the `originalUrl` field needs to be updated? Can you verify that our docs on all URL-related request fields is accurate? Also, if what we find to be true about Express (and thus mimic and write in our docs) differs from their docs, we should call it out in our docs on that field. For example, maybe this comment should be in the docs we expose for `originalUrl`.

Reply:
Also, in reading your tests, it seems like the Express docs are _somewhat right_ ... the `originalUrl` starts off as `baseUrl + path`; it's just that a rewrite of the `url` field can cause them to differ. Is that correct? If so, that should be noted.

Reply:
Yes, that's correct. Good idea, I'll update the docs, and double check that the rest of the URL-related request field docs are correct.